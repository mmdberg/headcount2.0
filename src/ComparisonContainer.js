import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './styles/comparisonContainer.css';

const ComparisonContainer = ({selectedCards, comparison, selectCard, deselectCard}) => {
  let newDistricts = [];
  let newDistrictValues = [];
  if (comparison) {
    newDistricts = Object.keys(comparison);
    newDistrictValues = Object.values(comparison);
  }
  const comparisonCards = selectedCards.map((district) => {
    return (
      <Card location={district.location}
        stats={district.stats}
        key={district.location}
        selectCard={selectCard}
        deselectCard={deselectCard}
        className="card clicked"
      />);
  });
  return (
    <div className="comparison-container">
      {comparisonCards[0]}

      {selectedCards.length === 2 &&
        <div className="comparison-card">
          <h2>{newDistricts[0]}</h2>
          <p>{newDistrictValues[0]}</p>
          <h2>{newDistricts[2]}</h2>
          <p>{newDistrictValues[2]}</p>          
          <h2>{newDistricts[1]}</h2>
          <p>{newDistrictValues[1]}</p>
        </div>
      }
      {comparisonCards[1]}
    </div>
  );
};

ComparisonContainer.propTypes = {
  selectedCards: PropTypes.array,
  comparison: PropTypes.array,
  selectCard: PropTypes.func,
  deselectCard: PropTypes.func
};


export default ComparisonContainer;