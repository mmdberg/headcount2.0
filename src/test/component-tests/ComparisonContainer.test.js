import React from 'react';
import { shallow, mount } from 'enzyme';
import mockCardContainerData from '../../data/mock_card_container_data'
import ComparisonContainer from '../../ComparisonContainer';

describe ('ComparisonContainer', () => {

  it('should match the snapshot', () => {

    let wrapper = shallow(<ComparisonContainer selectedCards={mockCardContainerData}/>)

    expect(wrapper).toMatchSnapshot();
  })

})