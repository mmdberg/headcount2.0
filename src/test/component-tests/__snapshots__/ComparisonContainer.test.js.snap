// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    selectedCards={
        Array [
            2004,
            2005,
            2006,
            2007,
            2008,
            2009,
            2010,
            2011,
            2012,
            2013,
            2014,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        false,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          false,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
